src/components/ui/Link/Link.tsx

```diff
@@ -1,36 +1,57 @@
+import { forwardRef } from 'react'
+import type { Ref, ElementType, AnchorHTMLAttributes } from 'react'
+import NextLink from 'next/link'
+import type { LinkProps as FrameworkLinkProps } from 'next/link'
 import { Link as UILink } from '@faststore/ui'
-import type { ElementType } from 'react'
 import type { LinkProps as UILinkProps } from '@faststore/ui'
 
-import FrameworkLink from 'src/components/common/Link'
-
 import styles from './link.module.scss'
 
+function isExternalLink(href: string) {
+  return (
+    href.startsWith('//') ||
+    href.startsWith('http://') ||
+    href.startsWith('https://')
+  )
+}
+
 type Variant = 'default' | 'display' | 'footer' | 'inline'
 
-export type LinkProps<T extends ElementType = typeof FrameworkLink> =
-  UILinkProps<T> & {
+export type LinkProps<T extends ElementType = 'a'> = UILinkProps<T> &
+  FrameworkLinkProps &
+  AnchorHTMLAttributes<HTMLAnchorElement> & {
     variant?: Variant
     inverse?: boolean
   }
 
-function Link<T extends ElementType = typeof FrameworkLink>({
-  variant = 'default',
-  inverse,
-  href,
-  ...otherProps
-}: LinkProps<T>) {
+const Link = forwardRef<HTMLAnchorElement, LinkProps>(function Link<
+  T extends ElementType = 'a'
+>(
+  { href, inverse, children, variant = 'default', ...otherProps }: LinkProps<T>,
+  ref: Ref<HTMLAnchorElement> | undefined
+) {
+  const defaultProps = {
+    'data-fs-link': true,
+    'data-fs-link-variant': variant,
+    'data-fs-link-inverse': inverse,
+    className: styles.fsLink,
+  }
+
+  if (isExternalLink(href)) {
```

Comment:
I feel like we could benefit from `useMemo` in here:
```suggestion
  const external = useMemo(() => isExternalLink(href), [href])
  if (external) {
```